packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+import '../material/material.dart'; // non ideal
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second
+
+// Used for iOS.
+class CupertinoPageTransition extends AnimatedWidget {
+  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
+    begin: FractionalOffset.topRight,
+    end: -FractionalOffset.topRight
+  );
+
+  CupertinoPageTransition({
+    Key key,
+    Animation<double> animation,
+    this.child
+  }) : super(
+    key: key,
+    listenable: _kTween.animate(new CurvedAnimation(
+      parent: animation,
+      curve: new _CupertinoTransitionCurve(null)
+    )
+  ));
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    // but not while being controlled by a gesture.
+    return new SlideTransition(
+      position: listenable,
+      child: new Material(
```

Comment:
Maybe we should use PhysicalModel. If we look at the native iOS behaviour carefully, I think this has to know about the elevation since it actually animates the elevation too.